jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -1074,6 +1074,27 @@ def test_errors(self):
         "Shapes must be 1D sequences of concrete values of integer type"):
       core.dimension_as_value(np.float32(1))
 
+  def test_error_jax_value(self):
+
+    x = np.ones([3, 5], dtype=np.float32)
+
+    with self.assertRaisesRegex(
+        core.InconclusiveDimensionOperation,
+        "Dimension polynomial 'b' used in a context that requires a constant"):
+      self.CheckShapePolymorphism(
+          # Cannot use dimension variable as a JAX value
+          lambda x: jnp.array(x.shape[0]),
+          input_signature=[tf.TensorSpec([None, 5], dtype=x.dtype)],
+          polymorphic_shapes=[("b, _")],
+          expected_output_signature=tf.TensorSpec([]))
+
+    self.CheckShapePolymorphism(
+        # We can convert a dimension variable to a JAX value
+        lambda x: jnp.array(core.dimension_as_value(x.shape[0])),
```

Comment:
Why calling `core.dimension_as_value` could work? Is this a public API we expect JAX users to use?

Reply:
`core.dimension_as_value` is right now an undocumented API, and it is used internally. We may remove the need for it. 

What it does is wrap a dimension as a JAX value that can then be used in JAX computations. The classic need for it is when implementing `jnp.mean` as `jnp.sum(x, axis=0) / x.shape[0]`. You need to wrap `x.shape[0]` with `core.dimension_as_value` or else you will get an error when the first dimension is polymorphic, because you cannot divide a JAX value by a dimension polynomial.